frontend/src/app/map-view/calculation-history/calculation-history.component.html

```diff
@@ -29,45 +41,52 @@ <h3>{{ 'map.history.previous-calculations' | translate }}</h3>
     ></app-list-filter>
   </div>
 
-  <main *ngIf="!(loading$ | async) else loading" >
-    <button *ngFor="let calculation of calculations$ | async; index as i"
-            [ngClass]="{
-              'calculation list-item' : true,
-              'filtered': filter(calculation.name),
-              'visible': resultIsVisible(calculation.id),
-              'purged': calculation.isPurged }"
-            [title]="(calculation.isPurged ? ('map.history.purged-result' | translate) : '')"
-            (click)="showReport(calculation.id)"
-            >
-      <div class="calculation-info">
-        <div class="name">
-          <h4 *ngIf="editingName !== calculation.id" [title]="calculation.name">{{ calculation.name }}</h4>
-<!--
-          <input #name *ngIf="editingName === calculation.id" [value]="calculation.name"
-                 (keydown.enter)="saveName($event, calculation, i)" (blur)="cancelEdit($event)"
-                 (click)="$event.stopPropagation()"/>
--->
+  <main *ngIf="!(loading$ | async) else loading">
+    <div class="calculation-wrap" *ngFor="let calculation of calculations$ | async; index as i">
+      <button [ngClass]="{
+                'calculation list-item' : true,
+                'filtered': filter(calculation.name),
+                'visible': resultIsVisible(calculation.id),
```

Comment:
Could be a performance issue since the array is iterated for every render

Reply:
This, to be clear, as well wasn't introduced in the current PR, however I'm responsible for these additions (it relates to functionality introduced in c1ccc08, **SYM-585**) and I do think this is an appropriate implementation, given the requirements (refer to **SYM-584/585** internally).
To be specific, these properties really should be dynamic - the alternative would incur a much more severe performance cost for the component to 'hydrate' (i.e. reload from backend). Feel free to verify this!